src/main/java/teammates/common/datatransfer/questions/FeedbackRankQuestionDetails.java

```diff
@@ -64,10 +64,7 @@ public abstract String getQuestionWithoutExistingResponseSubmissionFormHtml(
     protected void updateOptionRanksMapping(
                         Map<String, List<Integer>> optionRanks,
                         String optionReceivingRanks, int rankReceived) {
-        if (!optionRanks.containsKey(optionReceivingRanks)) {
-            List<Integer> ranks = new ArrayList<>();
-            optionRanks.put(optionReceivingRanks, ranks);
-        }
+        optionRanks.putIfAbsent(optionReceivingRanks, new ArrayList<Integer>());
 
         List<Integer> ranksReceived = optionRanks.get(optionReceivingRanks);
```

Comment:
I tried the substitution of (putIfAbsent + get) into computeIfAbsent and all the parts passed the tests except this part. If I replace the above two lines into 
`List<Integer> ranksReceived = optionRanks.computeIfAbsent(optionReceivingRanks, (String key) -> new ArrayList<>());`
FeedbackSessionLogicTest will fail and produce the following result: 
![_20171224005144](https://user-images.githubusercontent.com/28526038/34321236-d6257348-e844-11e7-942a-58d1b39b95f8.png)

Can i get some help on understanding why is this so? Thx!

Reply:
@wenmogu I'll take a look when I get the time; meanwhile, if you fix this test case (i.e. change the expected to match the actual), does anything else break?

Reply:
@wkurniawan07 after i changed the expected to match the actual, nothing else breaks.

Reply:
@wenmogu Have you figured out what caused the ordering issue?

Reply:
@wenmogu I think we are safe to change the test case.
Refer to https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java#L310

We are just enumerating all entries without any order (and therefore the order would depends on the internal implementation of `Map`, in this case `HashMap`)

You can update the branch and I will give my review. :)

Reply:
@xpdavid Thx for the help! Indeed i was struggling for a long time to figure out why this happens. I just updated the branch and pls review it:)